### DISCUSSION TITLE
[Proposal]: Action lambda assignment syntax sugar

### DISCUSSION THREAD

**Discussion by pedro-gilmora** [dotnet/csharplang#7486] [Language Ideas]
# Action lambda assignment syntax sugar

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary
[summary]: #summary

It's just a smaller way to assign lambda to `Action` reference variables
```cs
Action action ==> _defaultActionImplementation?.Invoke();
//Even better incrementing invoke list
action +=> extraAction();
```

## Motivation
[motivation]: #motivation

Just to make lighter syntax

**Comment by HaloFour** [comment]
Just to clarify:

```cs
Action action ==> _defaultActionImplementation?.Invoke();

// is equivalent to
Action action = () => _defaultActionImplementation?.Invoke();
// or
Action action = _defaultActionImplementation != null ? _defaultActionImplementation.Invoke : null;
```

And
```cs
action +=> extraAction();

// is equivalent to
action += () => extraAction();
// or
action += extraAction;
```

**Reply by pedro-gilmora** [reply]
the body is not important. It might be 

```cs
Action action ==> { 
  _defaultActionImplementation?.Invoke();
  InitialComputing(initialValue);
};
```
or
```cs
action +=> { 
  var result = ExtraComputing(newValue);
  Add(result);
};
```

It just should to add `()` between `=` or `+` and `=>` in the lowered version

**Comment by 333fred** [comment]
As noted in our [README](https://github.com/dotnet/csharplang/blob/main/README.md), new topics should be opened as discussions, not issues.
